website/docs/advanced-topics/server-side-rendering.md

```diff
@@ -126,11 +126,67 @@ suspended.
 With this approach, developers can build a client-agnostic, SSR ready
 application with data fetching with very little effort.
 
+## SSR Hydration
+
+Hydration is the process that connects a Yew application to the
+server-side generated HTML file. By default, `ServerRender` prints
+hydratable html string which includes additional information to facilitate hydration.
+When the `Renderer::hydrate` method is called, instead of start rendering from
+scratch, Yew will reconcile the Virtual DOM generated by the application
+with the html string generated by the server renderer.
+
+:::caution
+
+To successfully hydrate an html representation created by the
+`ServerRenderer`, the client must produce a Virtual DOM layout that
+exactly matches the one used for SSR including components that do not
+contain any elements. If you have any component that is only useful in
+one implementation, you may want to use a `PhantomComponent` to fill the
+position of the extra component.
+:::
+
+## Component Lifecycle during hydration
+
+During Hydration, components schedule 2 consecutive renders after it is
+created. Any effects are called after the second render completes.
+It is important to make sure that the render function of the your
+component is side-effect free. It should not mutate any states or trigger
+additional renders. If your component currently mutates states or triggers
+additional renders, move them into an `use_effect` hook.
+
+It's possible to use Struct Components with server-side rendering in
+hydration, the view function will be called
+multiple times before the rendered function will be called.
+The DOM is considered as not connected until rendered function is called,
+you should prevent any access to rendered nodes
+until `rendered()` method is called.
+
+## Example
+
+```rust ,ignore
+use yew::prelude::*;
+use yew::Renderer;
+
+#[function_component]
+fn App() -> Html {
+    html! {<div>{"Hello, World!"}</div>}
+}
+
+fn main() {
+    let renderer = Renderer::<App>::new();
+
+    // hydrates everything under body element, removes trailing
+    // elements (if any).
+    renderer.hydrate();
+}
+```
+
 Example: [simple\_ssr](https://github.com/yewstack/yew/tree/master/examples/simple_ssr)
+Example: [ssr\_router](https://github.com/yewstack/yew/tree/master/examples/ssr_router)
```

Comment:
I would bring these in line with how it's placed in other places:

```mdx
## Further reading

- The [`simple_ssr`](https://github.com/yewstack/yew/tree/master/examples/simple_ssr) example
- The [`ssr_router`](https://github.com/yewstack/yew/tree/master/examples/ssr_router) example
```

It's not blocking and can be done in a future PR